org.eclipse.xtext/src/org/eclipse/xtext/build/Indexer.java

```diff
@@ -213,21 +221,32 @@ public URI getURI() {
 	protected List<IResourceDescription.Delta> getDeltasForDeletedResources(BuildRequest request,
 			ResourceDescriptionsData oldIndex, BuildContext context) {
 		List<IResourceDescription.Delta> deltas = new ArrayList<>();
-		for (URI deleted : request.getDeletedFiles()) {
-			IResourceServiceProvider resourceServiceProvider = context.getResourceServiceProvider(deleted);
-			if (resourceServiceProvider != null) {
-				operationCanceledManager.checkCanceled(context.getCancelIndicator());
-				IResourceDescription oldDescription = oldIndex != null ? oldIndex.getResourceDescription(deleted)
-						: null;
-				if (oldDescription != null) {
-					DefaultResourceDescriptionDelta delta = new DefaultResourceDescriptionDelta(oldDescription, null);
-					deltas.add(delta);
+		if (oldIndex != null) {
+			for (URI deleted : request.getDeletedFiles()) {
+				IResourceServiceProvider resourceServiceProvider = context.getResourceServiceProvider(deleted);
+				if (resourceServiceProvider != null) {
+					operationCanceledManager.checkCanceled(context.getCancelIndicator());
+					IResourceDescription.Delta delta = getDeltaForResourceToDelete(deleted, oldIndex);
+					if (delta != null) {
+						deltas.add(delta);
+					}
 				}
 			}
 		}
 		return deltas;
 	}
 
+	/**
+	 * Gets a delta for a resource that shall be deleted.
+	 */
+	protected IResourceDescription.Delta getDeltaForResourceToDelete(URI uri, ResourceDescriptionsData oldIndex) {
```

Comment:
Can we align the method names. Right now we have `getDeltasForDeletedResources` and `getDeltaForResourceToDelete`.
Also please add an `@Since` tag to the new method.